First three sections of 1st draft portfolio

Ninjaphil24 opened the pull request:
Good Afternoon, I hope you're well.

I'm using the 2nd template from Figma for this exercise.

In this exercise it states:
Build the first 3 sections of the mobile website using the template you chose in the previous step.
(https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/1_setup.md)

It is unclear to me if the exercise wants us to build the first three sections, as in nav, aside, section, or the first three pages.  

I have been at it today since 7am and have only just finished this first page after 4 hours.  I'm happy to do two more pages if that's what the exercise requires, but I'd like this clarified before I do. 

As for what I've built so far:

I'm getting the following linter error:

90:18  ✖  Unexpected missing generic    font-family-no-missing-generic-family-keyword 
            font family
 100:18  ✖  Unexpected missing generic    font-family-no-missing-generic-family-keyword 
            font family
 109:18  ✖  Unexpected missing generic    font-family-no-missing-generic-family-keyword 
            font family
g-generic-family-keyword
            font family
  99:16  ✖  Unexpected missing generic    font-family-no-missinss}"     g-generic-family-keyword
            font family
 108:16  ✖  Unexpected missing generic    font-family-no-missing-generic-family-keyword g-generic-family-keyword
            font family                                        g-generic-family-keyword 

I have attempted to understand how stylelint works, but I have found the following article:

https://github.com/stylelint/stylelint/issues/3848

It suggests that this is a bug.  If it is a bug, I'm not entirely sure what steps to follow in order to fix it.  If it's legitimate, again, I'm not clear on how to fix it.  Advice would be appreciated.


Concerning the use of the image in index.html line 39:
I found that using the vector images separately was extremely challenging and time consuming (the task says 2.5 hours and I've been at it for 4).  Furthermore, there was a lot of css info in there that we haven't covered, such as transform matrix.  As such, I opted to do a screenshot of the image and used that instead.  Further down the line, I will be happy to learn how to do the Vector images on top of one another, but I felt this was too advanced and time consuming for today's purposes.

Ninjaphil24 committed:
repo and linter set

Ninjaphil24 committed:
README 1st draft

Ninjaphil24 committed:
navbar basic positions

Ninjaphil24 committed:
navbar basic positions linter

Ninjaphil24 committed:
navbar final color and hover

Ninjaphil24 committed:
sidebar final color and hover

Ninjaphil24 committed:
add text, but with lint errors

Ninjaphil24 committed:
First page complete with linter errors

mmsesay commented on README.md:
```diff
@@ -1,2 +1,67 @@
+### Run tests
+No linter errors
+### Deployment
+Click on "index.html"
```
[OPTIONAL] Hello, good job providing a readme for your project. However, please feel free to personalize it and remove lines that aren't part of your project.

mmsesay commented on index.css:
```diff
@@ -0,0 +1,123 @@
+}
+
+.home .text h2 {
+  font-family: Crete Round;
```
- [ ] I think this is the reason why your stylint linter is failing.
 `Crete Round` is an external font. Therefore you need to associate a generic font with it.  And `serif` is a common generic font family.

 Try this out `font-family: 'Crete Round', serif;`

Please try the above in all lines where you used external fonts.

mmsesay commented on index.html:
```diff
@@ -0,0 +1,45 @@
+<body>
+<div class="page1">
+    <header>
+    <h1><a href="index.html">Philip Modinos</a></h1>
```
- [ ] Please remove the `h1` tag that surrounds the `a` tag

Ninjaphil24 replied:
Hello again, I'll do it but I would like to know why.

mmsesay commented on index.html:
```diff
@@ -0,0 +1,45 @@
+    <h1><a href="index.html">Philip Modinos</a></h1>
+    <nav>
+        <ul>
+            <li><a href="portfolio.html">Portfolio</a></li>
+            <li><a href="about.html">About</a></li>
+            <li><a href="contact.html">Contact</a></li>
```
- [ ] For this task you only need an [hamburger](https://fontawesome.com/v5.15/icons/bars?style=solid) icon here.
Let's try to replace the `li` tags.

mmsesay commented on index.html:
```diff
@@ -0,0 +1,45 @@
+    <link href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-wvfXpqpZZVQGK6TAh5PVlGOfQNHSoD2xbE+QkPxCAFlNEevoEH3Sl0sibVcOQVnN" crossorigin="anonymous">
+</head>
+<body>
+<div class="page1">
+    <header>
+    <h1><a href="index.html">Philip Modinos</a></h1>
+    <nav>
+        <ul>
+            <li><a href="portfolio.html">Portfolio</a></li>
+            <li><a href="about.html">About</a></li>
+            <li><a href="contact.html">Contact</a></li>
+        </ul>
+    </nav>
+    </header>
+
+    <aside>
+        <ul>
+            <li><a href="https://github.com/Ninjaphil24"><i class="fa fa-github fa-3x" aria-hidden="true"></i></a></li>
+            <li><a href="https://twitter.com/PModinos"><i class="fa fa-twitter-square fa-3x" aria-hidden="true"></i></a></li>
+            <li><a href="https://www.linkedin.com/in/philip-modinos-14195021"><i class="fa fa-linkedin-square fa-3x" aria-hidden="true"></i></a></li>
+        </ul>
+    </aside>   
+
+    <section class="home">
+        <div class="text">
+        <h2>Hey There. <br> I'm Phil</h2>
+        <h3>I'm a Software Developer</h3>
+        <p>I can help you build a product , feature or website Look through some of my work and experience! If you like what you see and have a project you need coded, 
+            don’t hestiate to contact me.</p>
+        </div>
+        <div class="image">
+            <img id="Vector" src="images/fullimage.png" alt="">
+        </div>
+    </section>
+</div>
```
- [ ] It will be nicer if to follow the design from figma. Since you are using template 2 follow the layout, font family,  positioning etc.

Your Design | Figma Design |
:---------------:|:-------------------:|
![a2](https://user-images.githubusercontent.com/39523312/148217600-38c3dc9c-cbc1-4242-91d7-18935b59c477.png) | ![a1](https://user-images.githubusercontent.com/39523312/148217614-24d00a66-15ef-4a4e-a14e-29e8e57203c0.png)

Ninjaphil24 replied:
Hello Muhammad, thank you for your comments.  I'm going to do all that.  In order not to lose what I've done, I'll use a mediaquery to do the mobile version.

Ninjaphil24 replied:
Hello again,

I have resolved all the issues, I hope.
I've done something rather complicated, in order not to lose my desktop version, which I worked on for about 4 hours.  

I've created a mobile version using media queries and I've created two versions of that, as when it gets very tight, the image doesn't look so good, so I make it disappear.  

Once the project progresses and javascript is used for the Hamburger menu I will change the media queries to be something simpler.

mmsesay reviewed (changes requested):
Hi @Ninjaphil24 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Please don't get confused, you should only build the `toolbar` and `headline` sections of your chosen design. Try not to go beyond that.
I noticed you work on the desktop version which is good but not required for today.
Kindly follow [Template 2 design](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=1%3A1471) in building the mobile version. 
I placed a screenshot of your current design and the figma template below. You only need to work on the first 2 sections of the mobile menu.

I hope you get the point I am making here.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Ninjaphil24 committed:
Removed excess README info and repaired linter issues

Ninjaphil24 committed:
Final version with media queries for mobile

Ninjaphil24 committed:
minor linter problems fixed

Ninjaphil24 committed:
Forgot to remove borderline guides. Removed them now

elmar8287 commented on index.html:
```diff
@@ -0,0 +1,57 @@
+    <a href="index.html">Philip Modinos</a>
+    <nav class="desktop">
+        <ul>
+            <li><a href="portfolio.html">Portfolio</a></li>
+            <li><a href="about.html">About</a></li>
+            <li><a href="contact.html">Contact</a></li>
```
- [ ] Please, consider the comment from the previous review, to use the hamburger icon instead of `li` tags.

Ninjaphil24 replied:
I have used it.  If you notice, the media query under 1000px makes the li disappear.  I just didn't want to erase it, because later I'll probably need it for the desktop version.

elmar8287 reviewed (changes requested):
Hi @Ninjaphil24, 

Well done on making the suggested changes. But there is something to do before merging.

Required Changes ♻️

- Check my comment under the review.
- Please, make your README file more informative, refer MISCROVERSE template.
- Please, fix the screenshot in the README file, it is not displaying correctly.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Ninjaphil24 committed:
Final corrections

Ninjaphil24 commented:
Hello,

I have updated made the final corrections to my project.  Some **important** notes:

1. The 1st reviewer, Muhammad asked me to reduce the size of the README file.  However the second reviewer, Elmar, asked me to have more details.  It needs to be noted that I used the template and erased all sections that remained blank.  There is a conflict here in the views of the two reviewers.  I have added the screenshot as Elmar suggested, however, I will not add the stuff I erased, as it doesn't add any information at this time.  
2. Because I accidentally did the desktop version and not the mobile, I decided to use media queries to get the mobile version working.  This means that I've created a query that makes the li tags disappear when width < 1000px in index.css line 137.  I don't want to erase any of the work that I did on the desktop version, as I will probably need it later, when javascript is introduced.  The hamburger menu icon appears correctly when width <1000px and therefore, I'd like to leave the li tags as they are.

Thanks for all the help!!

emmanuellekamwa reviewed (approved):
**STATUS: APPROVED 🎉🍾✔**
Hello @Ninjaphil24 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
